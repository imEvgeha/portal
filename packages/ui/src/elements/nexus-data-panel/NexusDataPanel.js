import React from 'react';
import PropTypes from 'prop-types';
import {BlockUI} from 'primereact/blockui';
import {ProgressSpinner} from 'primereact/progressspinner';
import './NexusDataPanel.scss';

const NexusDataPanel = ({header, data, itemTemplate, footer, loading, contentFooter}) => {
    const defaultItemTemplate = item => {
        return (
            <div className="row">
                <div className="col-12">{item}</div>
            </div>
        );
    };

    const renderItems = () => {
        if (Array.isArray(data)) {
            return data.map(item => {
                if (itemTemplate) {
                    return itemTemplate(item);
                }
                return defaultItemTemplate(item);
            });
        }
    };

    return (
        <div className="nexus-c-data-panel h-100 d-flex flex-column">
            {header && (
                <div className="row nexus-c-header mb-2">
                    <div className="col-12">{header}</div>
                </div>
            )}

            <div className={`row nexus-c-content ${loading ? '' : 'nexus-c-overflow'}`}>
                <div className="col-12">
                    <BlockUI blocked={loading}>{renderItems()}</BlockUI>
                    {!!contentFooter && contentFooter}
                    {loading && (
                        <div className={Array.isArray(data) && data.length ? 'nexus-c-loading-wrapper' : 'text-center'}>
                            <ProgressSpinner
                                className="nexus-c-data-panel__spinner"
                                strokeWidth="4"
                                animationDuration=".5s"
                            />
                        </div>
                    )}
                </div>
            </div>

            {!!footer && footer}
        </div>
    );
};

NexusDataPanel.propTypes = {
    header: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    data: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
    itemTemplate: PropTypes.func,
    footer: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    contentFooter: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    loading: PropTypes.bool,
};

NexusDataPanel.defaultProps = {
    header: '',
    data: [],
    itemTemplate: undefined,
    footer: undefined,
    contentFooter: undefined,
    loading: false,
};

export default NexusDataPanel;
