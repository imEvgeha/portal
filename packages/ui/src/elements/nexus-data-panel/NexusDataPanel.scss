.nexus-c-data-panel {
    position: relative;
    .nexus-c-content {
        &.nexus-c-overflow {
            overflow-y: auto;
            overflow-x: hidden;
        }

        .nexus-c-data-panel__spinner {
            width: 50px;
            height: 50px;
        }

        .nexus-c-loading-wrapper {
            position: absolute;
            top: 55%;
            left: calc(50% - 50px);
        }
    }
}
