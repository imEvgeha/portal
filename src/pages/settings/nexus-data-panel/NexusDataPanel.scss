.nexus-c-data-panel {
    .content {
        overflow-y: auto;
        overflow-x: hidden;
    }
}
