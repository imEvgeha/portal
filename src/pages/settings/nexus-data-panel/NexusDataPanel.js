import React from 'react';
import PropTypes from 'prop-types';
import {ScrollPanel} from 'primereact/scrollpanel';
import './NexusDataPanel.scss';

const NexusDataPanel = ({header, data, itemTemplate, footer}) => {
    const defaultItemTemplate = item => {
        return (
            <div className="row">
                <div className="col-12">{item}</div>
            </div>
        );
    };

    const renderItems = () => {
        if (Array.isArray(data)) {
            return data.map(item => {
                if (itemTemplate) {
                    return itemTemplate(item);
                }
                return defaultItemTemplate(item);
            });
        }
    };

    return (
        <div className="nexus-c-data-panel h-100 d-flex flex-column">
            {header && (
                <div className="row header">
                    <div className="col-12">{header}</div>
                </div>
            )}

            <div className="row content">
                <div className="col-12">{renderItems()}</div>
                <div>LOAD MORE</div>
            </div>

            {footer && (
                <div className="row footer">
                    <div className="col-12">
                        <div>{footer}</div>
                    </div>
                </div>
            )}
        </div>
    );
};

NexusDataPanel.propTypes = {
    header: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    data: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
    itemTemplate: PropTypes.func,
    footer: PropTypes.string,
};

NexusDataPanel.defaultProps = {
    header: '',
    data: [],
    itemTemplate: undefined,
    footer: '',
};

export default NexusDataPanel;
