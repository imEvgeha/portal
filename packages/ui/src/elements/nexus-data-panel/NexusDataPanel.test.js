import React from 'react';
import {shallow} from 'enzyme';
import * as Redux from 'react-redux';
import NexusDataPanel from './NexusDataPanel';

describe('NexusDataPanel', () => {
    let wrapper = null;
    beforeEach(() => {
        wrapper = shallow(<NexusDataPanel />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render wrapper class', () => {
        expect(wrapper.find('.nexus-c-data-panel')).toHaveLength(1);
    });
});
