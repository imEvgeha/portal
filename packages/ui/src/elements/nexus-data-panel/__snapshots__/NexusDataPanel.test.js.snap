// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NexusDataPanel should match snapshot 1`] = `
<div
  className="nexus-c-data-panel h-100 d-flex flex-column"
>
  <div
    className="row nexus-c-content pb-1 nexus-c-overflow"
  >
    <div
      className="col-12"
    >
      <BlockUI
        __TYPE="BlockUI"
        autoZIndex={true}
        baseZIndex={0}
        blocked={false}
        className={null}
        fullScreen={false}
        id={null}
        onBlocked={null}
        onUnblocked={null}
        style={null}
        template={null}
      />
    </div>
  </div>
</div>
`;
